### PR TITLE
Expose signing algorithm and per-file metadata

### PR DESCRIPTION
**Pull request by tofay**
*Description of changes:*
This change updates the PackageInfo struct to expose the algorithm used to sign the RPM, and per file information from the RPM.

Note I've removed the BaseNames, DirIndexes and DirNames fields from PackageInfo as they are superseded by the File information (I think their only use is to derive file information). Let me know if I should add those back in for back-compatibility.

I can also split this PR into separate signing algorithm/file info PRs if you prefer.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice. I expect @wagoodman's confirmation is also required? (This contribution is a rebasing of their commits in https://github.com/anchore/go-rpmdb, which is a fork used in https://github.com/anchore/syft. I'm a syft and trivy user that is trying to remove the need for that fork to resolve https://github.com/anchore/syft/issues/469 😄 )

### CODE REVIEW

**Comment by wagoodman** [comment]
@tofay no permission needed, but thank you for asking -- we've been wanting to get our fork synced upstream (and deleted) this is a huge help 🙏 !

File: pkg/rpmtags.go [hunk 1]
```diff
@@ -10,18 +10,25 @@ const (
...
+	RPMTAG_DIRINDEXES     = 1116 /* i[] */
+	RPMTAG_BASENAMES      = 1117 /* s[] */
+	RPMTAG_DIRNAMES       = 1118 /* s[] */
+	RPMTAG_FILESIZES      = 1028 /* i[] */
+	RPMTAG_FILEMODES      = 1030 /* h[] , specifically []uint16 (ref https://github.com/rpm-software-management/rpm/blob/2153fa4ae51a84547129b8ebb3bb396e1737020e/lib/rpmtypes.h#L53 )*/
+	RPMTAG_FILEDIGESTS    = 1035 /* s[] */
+	RPMTAG_FILEFLAGS      = 1037 /* i[] */
+	RPMTAG_FILEUSERNAME   = 1039 /* s[] */
+	RPMTAG_FILEGROUPNAME  = 1040 /* s[] */
+	RPMTAG_FILEDIGESTALGO = 5011 /* i  */
```
Comment (knqyf263) [inline pkg/rpmtags.go:31]: nit: could you sort tags by the value?

**Review by knqyf263 — COMMENTED** [review]
The original idea of this library is to mimic the behavior of RPM as much as possible. This is because it makes it easier to keep up with upstream changes. Also, this library is basically for general usage. That is why I kept DirIndexes, etc and return installed files via function since `RPMTAG_INSTFILENAMES` is a virtual tag, not an actual field. From that perspective, I prefer having file attributes in `PackageInfo` as is and returning `FileInfo` through `Files()` or something like that. What do you think?

It is not so strong opinion. If you think my idea has a big penalty of performance or readability, I'm ok with the current implementation.

File: pkg/rpmtags.go (hunk 1 above)
Comment (tofay) [reply pkg/rpmtags.go]: done

**Comment by tofay** [comment]
I'm happy to with your suggestion (I think it makes it more readable), so I've gone with that.

Do we need to expose separate functions for getting FileInfos and just file paths?
I thought not (but don't feel strongly about it), and updated InstalledFiles to return FileInfos. Is that ok?

**Comment by knqyf263** [comment]
>I thought not (but don't feel strongly about it), and updated InstalledFiles to return FileInfos. Is that ok?

Can we rename `InstalledFiles` to `InstalledFileNames` and keep it so that we can comply with `RPMTAG_INSTFILENAMES`?

**Comment by tofay** [comment]
Thanks for the feedback - I've now done that so there is both an `InstalledFiles` and an `InstalledFileNames`

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
Thank you so much!!

**Pull request merged**